File: apis/AdopterAPI/Requests/PostAdopterRequest.cs
using CommonDTOLibrary.Models;

namespace AdopterAPI.Requests
{
    public class PostAdopterRequest
    {
        public string? UserName { get; set; }
        public string? PhoneNumber { get; set; }
        public string? Email { get; set; }
        public AddressDTO? Address { get; set; }
    }
}

File: client/lib/services/adopter/requests/post_adopter_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pet_share/services/generics/post_address_request.dart';
import 'package:pet_share/services/generics/post_user_request.dart';

part 'post_adopter_request.g.dart';

@JsonSerializable()
class PostAdopterRequest extends PostUserRequest {
  PostAdopterRequest(
      {this.firstName = '',
      this.lastName = '',
      String userName = '',
      String phoneNumber = '',
      String email = '',
      PostAddressRequest? address})
      : super(
          userName: userName,
          phoneNumber: phoneNumber,
          email: email,
          address: address,
        );

  String firstName;
  String lastName;

  @override
  Map<String, dynamic> toJson() => _$PostAdopterRequestToJson(this);
  factory PostAdopterRequest.fromJson(Map<String, dynamic> json) =>
      _$PostAdopterRequestFromJson(json);
}
